### PR TITLE
[netcore] Make System.Diagnostics.TraceSourceTests.DefaultTraceListenerClassTests.EntryAssemblyName_Null_NotIncludedInTrace Pass

### PR DESCRIPTION
**Pull request by MaximLipnin**
This PR fixes `System.Diagnostics.TraceSourceTests.DefaultTraceListenerClassTests.EntryAssemblyName_Null_NotIncludedInTrace` failure.
I added  private static `s_forceNullEntryPoint` field and public static `GetEntryAssembly` method to shared part of Assembly class. Also I added private static `GetEntryAssemblyInternal` wrapper method to Assembly.Mono.cs to invoke the related native logic.

The same should be done on dotnet/runtime side.

The related part in Assembly.CoreCLR.cs:
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Assembly.CoreCLR.cs#L107-L118

Fixes https://github.com/mono/mono/issues/14905

### CODE REVIEW

File: netcore/System.Private.CoreLib/src/System/Reflection/Assembly.Mono.cs [hunk 1]
```diff
@@ -46,7 +46,18 @@ public static Assembly GetExecutingAssembly()
 		public static extern Assembly GetCallingAssembly ();
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		public static extern Assembly GetEntryAssembly ();
+		internal static extern Assembly GetEntryAssemblyInternal ();
+
+		public static Assembly GetEntryAssembly()
```
Comment (marek-safar) [inline netcore/System.Private.CoreLib/src/System/Reflection/Assembly.Mono.cs]: This should go to shared version of Assembly.cs

**Comment by MaximLipnin** [comment]
@monojenkins build failed

**Comment by MaximLipnin** [comment]
@monojenkins build failed

**Comment by MaximLipnin** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
Upstream PR was approved, merging this one.

**Pull request merged**